src/auth/providers.js

```diff
@@ -41,35 +55,66 @@ function userPassProvider(auth) {
         .then(response => response.json());
     },
 
+    /**
+     * Request that the stitch server send another email confirmation
+     * for account creation.
+     *
+     * @param {String} email the email to send a confirmation email for
+     * @returns {Promise}
+     */
     sendEmailConfirm: (email) => {
-      const fetchArgs = common.makeFetchArgs('POST', JSON.stringify({email}));
+      const fetchArgs = common.makeFetchArgs('POST', JSON.stringify({ email }));
       fetchArgs.cors = true;
 
       return fetch(`${auth.rootUrl}/local/userpass/confirm/send`, fetchArgs)
         .then(common.checkStatus)
         .then(response => response.json());
     },
 
+    /**
+     * Sends a password reset request to the stitch server
+     *
+     * @param {String} email the email of the account to reset the password for
+     * @returns {Promise}
+     */
     sendPasswordReset: (email) => {
-      const fetchArgs = common.makeFetchArgs('POST', JSON.stringify({email}));
+      const fetchArgs = common.makeFetchArgs('POST', JSON.stringify({ email }));
       fetchArgs.cors = true;
 
       return fetch(`${auth.rootUrl}/local/userpass/reset/send`, fetchArgs)
         .then(common.checkStatus)
         .then(response => response.json());
     },
 
-    passwordReset: (tokenId, token) => {
-      const fetchArgs = common.makeFetchArgs('POST', JSON.stringify({tokenId, token}));
+    /**
+     * Use information returned from the stitch server to complete the password
+     * reset flow for a given email account, providing a new password for the account.
+     *
+     * @param {String} tokenId the tokenId provided by the stitch server
+     * @param {String} token the token provided by the stitch server
+     * @param {String} password the new password requested for this account
+     * @returns {Promise}
+     */
+    passwordReset: (tokenId, token, password) => {
+      const fetchArgs =
+        common.makeFetchArgs('POST', JSON.stringify({ tokenId, token, password }));
       fetchArgs.cors = true;
 
       return fetch(`${auth.rootUrl}/local/userpass/reset`, fetchArgs)
         .then(common.checkStatus)
         .then(response => response.json());
     },
 
+
+    /**
+     * Initiate an email signup flow
```

Comment:
Add: "Will trigger an email to the requested account containing a link with the token and tokenId that must be returned to the server using `emailConfirm()` to activate the account."